### DISCUSSION TITLE
Help on retrieving train_score for all hypertuning iteration from tuning_object

### DISCUSSION THREAD

**Discussion by mfiguere** [pycaret/pycaret#3617] [Q&A]
Hi everyone,

Using pycaret version==3.0.2 I have been trying to retrieve the training scores from tuning object(randomized_search_cv), but it is not available, even though I has explicitly set return_train_score as True. It used to work on previous versions of pycaret (last year)

Can someone help on how to get this information?
Example below, cheers

```
from pycaret import classification as clf
exp = clf.setup(data = df_eda_pandas[use_cols], target = target_col, session_id=123, html=True) 
model_knn = clf.create_model('knn', fold=3,return_train_score=True)

grid_params = { 'n_neighbors' : [5,7,9,11,13,15],
               'weights' : ['uniform','distance'],
               'metric' : ['minkowski','euclidean','manhattan']}
obj_tuned_knn = clf.tune_model(estimator=model_knn, custom_grid=grid_params,return_tuner=True, tuner_verbose=False,return_train_score=True,fold=3, n_iter=2)

#when printing the cv_results train score not available
print(obj_tuned_knn[1].cv_results_)

```

The print results for the  best model seems ok, it shows CV-Val and CV-train 
![Screenshot from 2023-06-23 10-39-15](https://github.com/pycaret/pycaret/assets/27901942/21c4bee4-5fd7-4baf-8de9-0406c051396c)

but the final object contains only test score, obj_tuned_knn[1].return_train_score also shows False.
![Screenshot from 2023-06-23 10-45-31](https://github.com/pycaret/pycaret/assets/27901942/181f874f-586e-4f87-b3e1-021b104c7b33)

 {'mean_fit_time': array([0.1428113 , 0.16197364, 0.17604844, 0.18151609, 0.17005626]),
 'std_fit_time': array([0.0176154 , 0.02643089, 0.01808566, 0.01048425, 0.02460967]),
 'mean_score_time': array([0.11152514, 0.        , 0.09752194, 0.        , 0.        ]),
 'std_score_time': array([0.02182978, 0.        , 0.01923114, 0.        , 0.        ]),
 'param_actual_estimator__solver': masked_array(data=['liblinear', 'lbfgs', 'lbfgs', 'lbfgs', 'lbfgs'],
              mask=[False, False, False, False, False],
        fill_value='?',
             dtype=object),
 'param_actual_estimator__penalty': masked_array(data=['l2', 'l1', 'l2', 'l1', 'l1'],
              mask=[False, False, False, False, False],
        fill_value='?',
             dtype=object),
 'param_actual_estimator__C': masked_array(data=[545.5594781168514, 3792.690190732246,
                    4.281332398719396, 0.004832930238571752,
                    78.47599703514607],
              mask=[False, False, False, False, False],
        fill_value='?',
             dtype=object),
 'params': [{'actual_estimator__solver': 'liblinear',
   'actual_estimator__penalty': 'l2',
   'actual_estimator__C': 545.5594781168514},
  {'actual_estimator__solver': 'lbfgs',
   'actual_estimator__penalty': 'l1',
   'actual_estimator__C': 3792.690190732246},
  {'actual_estimator__solver': 'lbfgs',
   'actual_estimator__penalty': 'l2',
   'actual_estimator__C': 4.281332398719396},
  {'actual_estimator__solver': 'lbfgs',
   'actual_estimator__penalty': 'l1',
   'actual_estimator__C': 0.004832930238571752},
  {'actual_estimator__solver': 'lbfgs',
   'actual_estimator__penalty': 'l1',
   'actual_estimator__C': 78.47599703514607}],
 'split0_test_score': array([0.53284672,        nan, 0.53448533,        nan,        nan]),
 'split1_test_score': array([0.52309297,        nan, 0.52681764,        nan,        nan]),
 'split2_test_score': array([0.53277712,        nan, 0.53084029,        nan,        nan]),
 'mean_test_score': array([0.52957227,        nan, 0.53071442,        nan,        nan]),
 'std_test_score': array([0.00458164,        nan, 0.00313159,        nan,        nan]),
 'rank_test_score': array([2, 3, 1, 4, 5], dtype=int32)}

**Comment by ngupta23** [comment]
`cv_results_` is the outcome from the sklearn object. It is not controlled by pycaret. You can refer to the available details here:

https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html

**Reply by mfiguere** [reply]
Hi, it seems like the tune_model function is not setting the return_train_score correctly. I passed return_train_score=True to the tune_model function.
![return_train_score_example](https://github.com/pycaret/pycaret/assets/27901942/d6601dc9-2822-48ce-aaf6-f7d8a38ba263)
